src/core/player.js

```diff
@@ -363,14 +363,14 @@ DM.provide('Player',
             case 'fullscreenchange': this.fullscreen = DM.parseBool(event.fullscreen); break;
             case 'controlschange': this.controls = DM.parseBool(event.controls); break;
             case 'volumechange': this.volume = parseFloat(event.volume); this.muted = DM.parseBool(event.muted); break;
+            case 'ad_start': this.adData = event.adData;
             case 'video_start':
-            case 'ad_start':
             case 'ad_play':
             case 'playing':
             case 'play': this.paused = false; break;
             case 'end': this.ended = true; break; // no break, also set paused
+            case 'ad_end': this.adData = null;
```

Comment:
Should we do the same on `error` event ? Have you verified that `ad_end` is dispatched when `ad_error` occurs ? (I am pretty sure yes but we need to check).

Reply:
Good point. 
I'd say it should be consistent with NPE implementation (e.g. when do we clean it in the player state?)

Reply:
after verification (thx Dam), `ad_end` is well dispatched after an error